test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -620,46 +618,61 @@ def __get_param_dictionary(self):
+        returns True on match or False otherwise None
+
+        :param param_dic: dictionary of cluster information from data passed
+                          to param list
+        :param_dic type: dic
+        :param test: test object
+        :test type: object
+
+        :returns True or False or None
         """
-        keylist = ['num_servers', 'num_moms', 'num_comms', 'num_clients']
         ts_requirements = {}
         tc_requirements = {}
         param_count = {}
-        if test is not None:
-            test_name = getattr(test.test, '_testMethodName', None)
-            if test_name is not None:
-                method = getattr(test.test, test_name, None)
-            if method is not None:
-                tc_requirements = getattr(method, REQUIREMENTS_KEY, {})
-                cls = method.im_class
-                ts_requirements = getattr(cls, REQUIREMENTS_KEY, {})
-                if not tc_requirements:
-                    if not ts_requirements:
-                        return None
-            eff_tc_req = get_effective_reqs(ts_requirements, tc_requirements)
+        shortname = (socket.gethostname()).split('.', 1)[0]
+        if test is None:
+            return None
+        test_name = getattr(test.test, '_testMethodName', None)
+        if test_name is not None:
+            method = getattr(test.test, test_name, None)
+        if method is not None:
+            tc_requirements = getattr(method, REQUIREMENTS_KEY, {})
+            cls = method.im_class
+            ts_requirements = getattr(cls, REQUIREMENTS_KEY, {})
+        if not tc_requirements:
+            if not ts_requirements:
+                return None
+        eff_tc_req = get_effective_reqs(ts_requirements, tc_requirements)
         for key in ['servers', 'moms', 'comms', 'clients']:
-            param_count['num_'+key] = len(param_dict[key])
+            param_count['num_'+key] = len(param_dic[key])
         for pk in param_count:
-            if pk in keylist:
-                if param_count[pk] < eff_tc_req[pk]:
-                    return False
-        if set(param_dict['moms']) & set(param_dict['servers']):
+            if param_count[pk] < eff_tc_req[pk]:
+                return False
+        if set(param_dic['moms']) & set(param_dic['servers']):
             if eff_tc_req['no_mom_on_server']:
                 return False
         else:
             if not eff_tc_req['no_mom_on_server']:
                 return False
-        if set(param_dict['comms']) & set(param_dict['servers']):
+        if set(param_dic['comms']) & set(param_dic['servers']):
             if eff_tc_req['no_comm_on_server']:
                 return False
         else:
             if not eff_tc_req['no_comm_on_server']:
                 return False
-        if set(param_dict['moms']) & set(param_dict['comms']):
+        comm_mom_list = set(param_dic['moms']) & set(param_dic['comms'])
+        if shortname in comm_mom_list:
```

Comment:
if comm_mom_list is None then this will error out, so I think you need to change this to:  "if comm_mom_list and if shortname in comm_mom_list":